how-to/offline_cpu.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+# Copyright (c) 2019 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# Description: Offline SOS CPUs except BSP before launch UOS
+
+for i in `ls -d /sys/devices/system/cpu/cpu[1-9]*`; do
+        online=`cat $i/online`
+        idx=`echo $i | tr -cd "[0-9]"`
+        echo cpu$idx online=$online
+        if [ "$online" = "1" ]; then
+                echo 0 > $i/online
+                while [ "$online" = "1" ]; do
+                        sleep 1
```

Comment:
Why (and why 1 second)?

Reply:
To offline a CPU 1 is written to the file. Once the CPU is offline, there are many things that will happen in the background like moving all the processes, interrupt from the CPU and so on. ACRN team has come up with this upper limit of 1s for all these background operations.

Please refer https://www.kernel.org/doc/html/v4.14/core-api/cpu_hotplug.html for more info.

Reply:
Sleeps are unreliable though: what if someone runs this script on a very slow and heavily overloaded system? 1 second may not be enough? Ideally, the script should be performing a timed wait on the number of CPUs dropping by the number of CPUs this script offlines.

Reply:
@vijaydhanraj - gah - yep. I wasn't looking at the entire script ;)